src/item-editor.jsx

```diff
@@ -31,6 +32,20 @@ var ItemEditor = React.createClass({
     },
 
     render: function() {
+        var questionSearchIndex = -1;
+        var answerAreaSearchIndex = -1;
+
+        if (this.props.searchIndex !== -1) {
+            var questionSearchCount = Util.countOccurences(this.props.question.content, this.props.searchString);
+            var answerAreaSearchCount = Util.countOccurences(this.props.answerArea.options.content, this.props.searchString);
+            
+            if (this.props.searchIndex < questionSearchCount) {
+                questionSearchIndex = this.props.searchIndex;
+            } else if (this.props.searchIndex < questionSearchCount + answerAreaSearchCount) {
+                answerAreaSearchIndex = this.props.searchIndex - questionSearchCount;
+            }
+        }
```

Comment:
It's unfortunate having all of this search specific logic here and in hint-editor and item-editor.  Maybe I can pass down some sort of search result offsets and try to move this logic out to search-and-replace-dialog.jsx.